datalad/auto.py

```diff
@@ -125,6 +126,8 @@ def _proxy_open_name_mode(self, origname, origfunc, *args, **kwargs):
         try:
             if self._in_open:
                 raise _EarlyExit("within open already")
+            if self._getting_repo_instance:
+                raise _EarlyExit("we are opening an annex repo - no files must be annexed")
```

Comment:
Yeah, still no luck figuring out how to reach this on my end, but these changes look safe to me (though I'm not really familiar with the inner workings of auto.py).

Thanks